test/unit/test-action.js

```diff
@@ -735,6 +735,55 @@ describe('Action method', () => {
     expect(inv.args['key1']).to.equal(11);
     expect(inv.source).to.equal(child);
   });
+
+  describe('macros', () => {
+    let actionMacroDef;
+    let expectedActionMap;
+
+    beforeEach(() => {
+      const actionMacroId = 'action-macro-id';
+      actionMacroDef = {
+        event: 'amp-action-event',
+        target: id,
+        method: 'method1',
+        args: {arg0: 0, arg1: 1},
+        str: `${actionMacroId}.method1`,
+      };
+      action.actionMacros_['action-macro-id'] = actionMacroDef;
+      // The expected action map used when generating the action invocation.
+      // Note the merged properties with the action macro defined above.
+      expectedActionMap = {
+        args: {'arg0': 0, 'arg1': 2},
+        event: 'tap',
+        method: 'method1',
+        str: 'tap:' + id + '.method1(arg1=2)',
+        target: id,
+      };
+      // A caller that references an action macro, providing only
+      // argument.
+      targetElement.setAttribute('on', `tap:${actionMacroId}(arg1=2)`);
```

Comment:
In test code, prefer duplicating literals (e.g. strings) to make it easier to read. Check out `go/tott-510`.